### PR TITLE
fix(Table): Fixed filtering in columns with children.

### LINKED ISSUE #30809: Table filters not working when the column has children.
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/embed/reset-filters-and-sorters-antd4161-forked-tnypx?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
Try to filter the Person column.

### What is expected?
The column filter activates and allows to reset the filter.

### What is actually happening?
The filter does not activate.

| Environment | Info |
|---|---|
| antd | 4.16.1 |
| React | 16.14.0 |
| System | Chrome |
| Browser | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36 |

---
It works if the column does not have children.

### LINKED ISSUE #31184: Table "filteredValue" on column with children columns does not work
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://stackblitz.com/edit/react-f58pgp](https://stackblitz.com/edit/react-f58pgp)

### Steps to reproduce
1. click on button "show_columns_1"

### What is expected?
-checkbox for option "FirstNameCol" should be checked,

-button "show_columns_1" should be a active

### What is actually happening?
-checkbox for option "FirstNameCol" is not checked,

-button "show_columns_1" is not active

| Environment | Info |
|---|---|
| antd | 4.16.6 |
| React | 17.0.0 |
| System | MacOS |
| Browser | Chrome |

---
Filter "show_columns_1" should work similarly like in case of column without children "show_columns_2".
Same bug applies to `defaultFilteredValue` prop.

### PR DESCRIPTION
**Pull request by moraispgsi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #30809
fix #31184

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 🐞 Fix Table `filteredValue` in nested columns.  |
| 🇨🇳 Chinese |  修复 Table 组合列上的筛选状态 `filteredValue` 不生效的问题。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ timeline: true
 - Table
   - 🐞 Fix Table.Summary missing border style when `sticky` and `bordered` enabled. [#30666](https://github.com/ant-design/ant-design/pull/30666)
   - 🐞 Fix column align with sorter and filter. [#30744](https://github.com/ant-design/ant-design/pull/30744)
+  - 🐞 Fix filtering in nested columns. [#30809](https://github.com/ant-design/ant-design/pull/30809) [@moraispgsi](https://github.com/moraispgsi)
```
Comment (afc163) [inline CHANGELOG.en-US.md:32]: 不需要修改 changelog，changelog 留在 PR 内容里。

Comment (yoyo837) [reply CHANGELOG.en-US.md:32]: I can't see the revert you said in the files change.

Comment (moraispgsi) [reply CHANGELOG.en-US.md:32]: https://github.com/moraispgsi/ant-design/commit/a48bfbf6387838eb3d676787114a498e97c2650e

Comment (moraispgsi) [reply CHANGELOG.en-US.md:32]: Not sure why the PR hasn't updated.

Comment (yoyo837) [reply CHANGELOG.en-US.md]: Forgot push?

Comment (moraispgsi) [reply CHANGELOG.en-US.md]: It was a sync problem. I think it's available now.

**Pull request merged**